File: src/python/cdb/common/objects/userInfo.py
#!/usr/bin/env python

"""
Copyright (c) UChicago Argonne, LLC. All rights reserved.
See LICENSE file.
"""


from cdbObject import CdbObject

class UserInfo(CdbObject):

    DEFAULT_KEY_LIST = [ 'id', 'username', 'firstName', 'lastName', 'middleName', 'email', 'description' ]

    def __init__(self, dict):
        CdbObject.__init__(self, dict)

    def getDefaultUserGroup(self):
        userGroupList = self.get('userGroupList', [])
        if len(userGroupList):
            return userGroupList[0]
        return None

File: tools/developer_tools/cdb_plugins/plugins/docManagament/java/objects/UserInfo.java
/*
 * Copyright (c) UChicago Argonne, LLC. All rights reserved.
 * See LICENSE file.
 */
package gov.anl.aps.cdb.portal.plugins.support.docManagament.objects;

import gov.anl.aps.cdb.common.objects.CdbObject;

/**
 *
 * @author djarosz
 */
public class UserInfo extends CdbObject {
    private String email; 
    private String fullName;
    private String username;
    private String badgeNo; 

    public String getEmail() {
        return email;
    }

    public String getFullName() {
        return fullName;
    }

    public String getUsername() {
        return username;
    }

    public String getBadgeNo() {
        return badgeNo;
    }        
}
